flytekit/extras/tasks/shell.py

```diff
@@ -80,6 +80,8 @@ def subproc_execute(command: typing.Union[List[str], str], **kwargs) -> ProcessR
         # Execute the command and capture stdout and stderr
         result = subprocess.run(command, **kwargs)
 
+        assert result.stderr == ""
```

Comment:
sometimes commands write debug information to stderr, so we should have a more resilient way of checking if a subprocess ran to completion.

Reply:
I wish they didn't but you're right.. unfortunately, subprocess.run doesn't capture PIPESTATUS it seems, only showing the return code for the last command. Do you have any thoughts? The only thing I can think of would be to check stderr for "command not found" or "error", but that's pretty awkward.

Reply:
Unfortunately I don't have a great solution. 

pipestatus is a bash feature, so this complicates things a little since we use `/bin/sh` as the [default shell](https://github.com/flyteorg/flytekit/blob/master/flytekit/extras/tasks/shell.py#L207) in `ShellTask` and depending on the OS that's not bash (e.g. on MacOS `/bin/sh` is symlinked to bash:
```
❯ uname -a
Darwin gondor 23.3.0 Darwin Kernel Version 23.3.0: Wed Dec 20 21:31:00 PST 2023; root:xnu-10002.81.5~7/RELEASE_ARM64_T6020 arm64

❯ readlink /private/var/select/sh
/bin/bash
```
).

A low-lift solution is to write a warning to the flytekit logs that we detected the use of a pipe in the command and that means we might end up not capturing the output of the command. That's not a great solution either, but at least we let the users know that they are in pipeland, which means they will have to handle it (maybe mention pipestatus?). wdyt?

Reply:
Yeah, I think that's the best solution given the current constraints. It is a bit of a corner case and I don't think we should be expected to handle *every* failure state. How does the latest commit look?